resourceLoader.js

```diff
@@ -8,124 +8,281 @@ function resourceLoader(options) {
-				var url = this;
-
-				$.ajax({
-					url: url,
-					dataType: 'script',
-					cache: true
-				}).done(function() {
-					jsCounter++;
-					checkIfComplete();
-				}).fail(function () {
-					console.warn('Error while loading: ' + url);
-					jsCounter++;
-					checkIfComplete();
-				});
+				if (_cache[queueObject['uniquePath']]) {
+					queueObject['promise'].resolve();
+					(_debug) ? console.log('Promise for ' + queueObject['uniquePath'] + ' resolved from cache.') : '';
+				} else {
+					// load js file
+					if (queueObject['type'] === 'js') {
+						cachePromise = $.ajax({
+							url: queueObject['uniquePath'],
+							dataType: 'script',
+							cache: true
+						}).always(function () {
+							queueObject['promise'].resolve();
+							_queue = _queue.filter(function (item) {
+								return item['uniquePath'] !== queueObject['uniquePath'];
+							});
+
+							(_debug) ? console.log('Promise for ' + queueObject['uniquePath'] + ' resolved.') : '';
+							checkIfComplete();
+						}).fail(function () {
+							console.warn('Error while loading: ' + queueObject['uniquePath']);
+						});
+					}
+					// load css file
+					if (queueObject['type'] === 'css') {
+						var $link = $('<link/>', {
+							rel: 'stylesheet',
+							type: 'text/css',
+							href: queueObject['uniquePath']
+						}).on('load', function () {
+							queueObject['promise'].resolve();
+							_queue = _queue.filter(function (item) {
+								return item['uniquePath'] !== queueObject['uniquePath'];
+							});
+
+							(_debug) ? console.log('Promise for ' + queueObject['uniquePath'] + ' resolved.') : '';
+							checkIfComplete();
+						}).on('error', function () {
+							console.warn('Error while loading: ' + url);
+							queueObject['promise'].resolve();
+							_queue = _queue.filter(function (item) {
+								return item['uniquePath'] !== queueObject['uniquePath'];
+							});
+
+							(_debug) ? console.log('Promise for ' + queueObject['uniquePath'] + ' resolved.') : '';
+							checkIfComplete();
+						});
+
+						$('head').append($link);
+						cachePromise = $.Deferred().resolve();
+					}
+
+					_cache[queueObject['uniquePath']] = cachePromise;
+				}
 			});
-		}
+		});
 	};
 
 	if (options && options.resources) {
-		loader.conditionsAllArray = loader.conditionsAllArray.concat(options.resources);
+		_loader.conditionsAllArray = _loader.conditionsAllArray.concat(options.resources);
 	} else {
-		$('[data-resources]').each(function() {
+		$('[data-resources]').each(function () {
 			var obj = {};
 			obj.resources = eval($(this).attr('data-resources'));
-			loader.conditionsAllArray.push(obj);
+			_loader.conditionsAllArray.push(obj);
 		});
 	}
 
-	if (loader.conditionsAllArray.length > 0) {
+	if (_loader.conditionsAllArray.length > 0) {
 		getResources();
 		loadResources();
 	} else {
 		$(window).trigger('resourcesReady');
 	}
 }
+
+// creates absolute path from any given string
+function absolutePath(urlString) {
```

Comment:
The logic for absolute path handling doesn't seem robust enough to me. If I run the demo page from http://localhost/conditional-resource-loader/demo/demo.html or from http://localhost:8080/demo/demo.html, in both cases all resources fail to load due to wrongly normalized paths.